semgrep/semgrep/util.py

```diff
@@ -119,7 +119,13 @@ def progress_bar(
     Return tqdm-wrapped iterable if output stream is a tty;
     else return iterable without tqdm.
     """
-    if file.isatty():
+    # Conditions:
+    # file.isatty() - only show bar if this is an interactive terminal
+    # len(iterable) > 1 - don't show progress bar when using -e on command-line. This
+    #   is a hack, so it will only show the progress bar if there is more than 1 rule to run.
+    # not DEBUG - don't show progress bar with debug
+    # not QUIET - don't show progress bar with quiet
+    if file.isatty() and len(list(iterable)) > 1 and not DEBUG and not QUIET:
```

Comment:
if you _actually_ get an iterable this will consume it (so on line 130, the iterable will actually appear to be empty)

You could change the type hint to accept a `List` as an argument instead of an iterable or do a little bit of magic to read the first two elements of the iterable, then put the iterable back together with `itertools.chain`.

You could also just convert it to a list once, than iterate over it.

But as written, this won't work if an iterable is actually passed as an argument.

Reply:
OK, cool. Didn't think about that!